modules/imgproc/src/hough.cpp

```diff
@@ -105,51 +105,69 @@ array of (rho, theta) pairs. linesMax is the buffer size (number of pairs).
+    Mat img = image.getMat();
+
     int i, j;
     float irho = 1 / rho;
 
     CV_Assert( img.type() == CV_8UC1 );
 
-    const uchar* image = img.ptr();
+    const uchar* _image = img.ptr();
     int step = (int)img.step;
     int width = img.cols;
     int height = img.rows;
 
+    int max_rho = (img.rows + img.cols);
+    int min_rho = -max_rho;
+
     if (max_theta < min_theta ) {
         CV_Error( CV_StsBadArg, "max_theta must be greater than min_theta" );
     }
     int numangle = cvRound((max_theta - min_theta) / theta);
-    int numrho = cvRound(((width + height) * 2 + 1) / rho);
+    int numrho = cvRound(((max_rho - min_rho) + 1) / rho);
 
-#if defined HAVE_IPP && IPP_VERSION_X100 >= 810 && !IPP_DISABLE_HOUGH
-    CV_IPP_CHECK()
+    if (type == CV_32FC2)
     {
-        IppiSize srcSize = { width, height };
-        IppPointPolar delta = { rho, theta };
-        IppPointPolar dstRoi[2] = {{(Ipp32f) -(width + height), (Ipp32f) min_theta},{(Ipp32f) (width + height), (Ipp32f) max_theta}};
-        int bufferSize;
-        int nz = countNonZero(img);
-        int ipp_linesMax = std::min(linesMax, nz*numangle/threshold);
-        int linesCount = 0;
-        lines.resize(ipp_linesMax);
-        IppStatus ok = ippiHoughLineGetSize_8u_C1R(srcSize, delta, ipp_linesMax, &bufferSize);
-        Ipp8u* buffer = ippsMalloc_8u_L(bufferSize);
-        if (ok >= 0) {ok = CV_INSTRUMENT_FUN_IPP(ippiHoughLine_Region_8u32f_C1R, image, step, srcSize, (IppPointPolar*) &lines[0], dstRoi, ipp_linesMax, &linesCount, delta, threshold, buffer);};
-        ippsFree(buffer);
-        if (ok >= 0)
+#if defined HAVE_IPP && IPP_VERSION_X100 >= 810 && !IPP_DISABLE_HOUGH
+        CV_IPP_CHECK()
         {
-            lines.resize(linesCount);
-            CV_IMPL_ADD(CV_IMPL_IPP);
-            return;
+            std::vector<Vec2f> _lines;
+            IppiSize srcSize = { img.cols, img.rows };
+            IppPointPolar delta = { rho, theta };
+            IppPointPolar dstRoi[2] = {{(Ipp32f) min_rho, (Ipp32f) min_theta},{(Ipp32f) max_rho, (Ipp32f) max_theta}};
+            int bufferSize;
+            int nz = countNonZero(img);
+            int ipp_linesMax = std::min(linesMax, nz*numangle/threshold);
+            int linesCount = 0;
+            _lines.resize(ipp_linesMax);
+            IppStatus ok = ippiHoughLineGetSize_8u_C1R(srcSize, delta, ipp_linesMax, &bufferSize);
+            Ipp8u* buffer = ippsMalloc_8u_L(bufferSize);
+            if (ok >= 0) {ok = CV_INSTRUMENT_FUN_IPP(ippiHoughLine_Region_8u32f_C1R, image, step, srcSize, (IppPointPolar*) &_lines[0], dstRoi, ipp_linesMax, &linesCount, delta, threshold, buffer);};
+            ippsFree(buffer);
+            if (ok >= 0)
+            {
+                lines.create(linesCount, 1, CV_32FC2);
+                Mat(linesCount, 1, CV_32FC2, &_lines[0]).copyTo(lines.getMat());
+                CV_IMPL_ADD(CV_IMPL_IPP);
+                return;
+            }
+            lines.assign(Mat(0,0,CV_32FC2));
```

Comment:
Since the existing code contained the code for clearing the output,I corrected it accordingly, but it certainly is not necessary.

Reply:
Problem here is, that empty Mat should not have any type information (OpenCV doesn't support standalone metadata).

Reply:
I see, I'm thinking to fix it for the following, but what about it?
lines.assign (Mat ());
And since empty Mat should not have a type, I think that it should return an error.